### DISCUSSION TITLE
Malwarebytes blocking upgrade - just reboot

### DISCUSSION THREAD

**Discussion by Andy2No** [christofmuc/KnobKraft-orm#97] [General]
I had a problem with Malwarebytes stopping me upgrading something recently (a boards definition, in the Arduino IDE), and I've now had it happen trying to intall version 1.13.1 of Orm, over the top of version 1.13.0, so this about what to do if that happens.

The solution is to just reboot, then run the installation again.  After quite a lot of googling and forum searching, I never found any other way.

In this case, it was proabably caused by me having Orm running when I tried to install the new version.  I closed it and tried again, but Malwarebytes (Premium) had locked KnobKraftOrm.exe so that it couldn't be upgraded to the new version.

It's possible this only affects Windows 7, and maybe that's the only circumstance it will happen in, but Malwarebytes has been more paranoid than usual, of late, so there's a chance it will happen again, and to somoeone else.

This is how the problem looked, while trying to install 1.13.1:

![1_13_1 update blocked by MBAM](https://user-images.githubusercontent.com/8021302/121232813-e00daf00-c889-11eb-8d0d-83fbfcb7d54d.gif)

Cancelling, then restarting the installation just repeated that message.

While trying to figure what out files Malwarebytes Service was still using, I looked at the current installation with an unlocking tool, called LockHunter, which showed it was locking KnobKraftOrm.exe.  LockHunter wasn't able to unlock it, which is generally the case:

![1_13_1 update blocked by MBAM - locking old exe](https://user-images.githubusercontent.com/8021302/121233233-57dbd980-c88a-11eb-83c7-606d942bc908.gif)

After rebooting, the file was no longer locked, so I was able to successfully install the new version over it.  It's possible that if you leave it too long after rebooting, it will get locked again, when that happens.

**Comment by christofmuc** [comment]
@Andy2No Thanks for the information! I'll keep it in the back of my head. Virus scanners can be such a nuisance to support... but you really shouldn't be running Windows 7 anymore, it is EOL since January 2020 and not deemed to be a secure OS anymore. Just saying ;-)

**Comment by Andy2No** [comment]
You're welcome.   There's a lot in common with Win 10, so it might affect other people, at some point.

I don't intend to ever get Win 10.  I've been meaning to migrate to Linux, but I'd still use Win 7 offline, for software that isn't available on Linux, that I'm used to.  I still keep a laptop with Win XP on, and it's surprising what can be got to run on it.  It doesn't get to go online, though.
